File: oo-workshop/single-responsibillity-principle/src/test/java/com/example/interfacesegregation/mobprogramming/problem/OnlineClientImplTest.java
package com.example.interfacesegregation.mobprogramming.problem;

import org.junit.Test;;

import static org.junit.Assert.assertTrue;

public class OnlineClientImplTest {
    @Test
    void acceptOnlineOrder(){
        //arrange
        OnlineClientImpl given = new OnlineClientImpl("Brandon Gilbert", "Salad", "6.99");

        //act
        String when = given.acceptOnlineOrder();

        //assert
        assertTrue(when.contains("Brandon Gilbert"));
        assertTrue(when.contains("Salad"));
    }

    @Test
    void payOnline(){
        //arrange
        OnlineClientImpl given = new OnlineClientImpl("Brandon Gilbert", "Salad","6.99");

        //act
        String when = given.payOnline();

        //assert
        assertTrue(when.contains("6.99"));
    }
}

File: oo-workshop/tsexercises/tests/unit/interfacesegregation/onlineClientImpl.test.ts
import {OnlineClientImpl} from '../../../src/interfacesegregation/onlineClientImpl'

describe('online client implementation', () => {
 
    it('should accept an online order', () => {
        //arrange
        const given: OnlineClientImpl = new OnlineClientImpl("Brandon Gilbert", "drain tile", "6.99")

        //act
        const when: string = given.acceptOnlineOrder()

        //assert
        expect(when).toContain("Brandon Gilbert")
        expect(when).toContain("drain tile")
    })


    it('should accept online payment', () => {
        //arrange
        const given: OnlineClientImpl = new OnlineClientImpl("Brandon Gilbert", "drain tile", "6.99")

        //act
        const when: string = given.payOnline()

        //assert
        expect(when).toContain("6.99")
    })
})

